core/Network/TLS/Extension.hs

```diff
@@ -466,6 +467,16 @@ decodeSignatureAlgorithms = runGetMaybe $ do
 
 ------------------------------------------------------------
 
+data PostHandshakeAuth = PostHandshakeAuth deriving (Show,Eq)
+
+instance Extension PostHandshakeAuth where
+    extensionID _ = extensionID_PostHandshakeAuth
+    extensionEncode _               = B.empty
+    extensionDecode MsgTClientHello = runGetMaybe (pure PostHandshakeAuth)
```

Comment:
I believe that `return` is safer than `pure` from the backward compatibility point of view.